include/boost/variant/variant.hpp

```diff
@@ -2200,7 +2235,8 @@ class variant
 
     bool empty() const BOOST_NOEXCEPT
     {
-        return false;
+        detail::variant::empty_check_visitor visitor;
```

Comment:
Use the `BOOST_VARIANT_NO_RECURSIVE_WRAPPER_POINTER_STEALING` macro here for the users that wish to have the old behavior.

Reply:
Done